### ISSUE TITLE
Non-Frozen Cols not matching Frozen Cols size

### ISSUE THREAD

**Issue by cmrn-rhi** [cidgoh/DataHarmonizer#66]
**Release 0.6.0:** The non-frozen column rows do not resize to match the contents/height of the frozen column ‘specimen collect sample ID’ rows. I know this might sound confusing so you can reference figures **[here](https://docs.google.com/document/d/1hPdWeGglWImQCKoYyPCkah-yJPDR7qq9e5rAr5RSJIc/edit#heading=h.yvgniajezf5y)** for further clarification.

**Comment by ivansg44** [comment]
Here's a gif of the bug popping in and out as you scroll through the example data:

![scrollingBug](https://user-images.githubusercontent.com/34198261/84957668-86c4d280-b0b0-11ea-88b2-5693929050b3.gif)

**Comment by ddooley** [comment]
This is such a tough one.  The dynamic drawing of rows in chunks of columns is responsible for it.  Handsongrid only looks at a viewport of columns, and renders them, but its rendering of far left column isn't in sync with the row heights of the viewport all the time.  There is one possible low-tech solution - Ivan can Handsongrid have its column viewport extended so that it covers the entire row?  Right now it probably has a column window of like 20.  If this is a parameter that can be extended to the entire # of columns, that would likely fix it.  Then horizontal scroll won't run into different row heights.

**Comment by ivansg44** [comment]
Yes, the viewport can be extended. But I believe it makes things sluggish last time I tried it.

**Comment by cmrn-rhi** [comment]
Is it feasible to have separate horizontal scrollbar that appears when the viewport contents extend the maximum length? 
![viewport scrollbar](https://user-images.githubusercontent.com/48695054/84961827-25096600-b0ba-11ea-9979-4a1646980b47.png)

**Comment by ivansg44** [comment]
Perhaps. I don't know if Handsontable offers that natively, but we could always manipulate the grid using JQuery or something. That's risky though, because you never know when the grid's elements may behave unexpectedly (Handson does a bunch of its own DOM manipulations). Maybe if there's a way to add a class to the frozen column via Handsontable.

We should also continue investigating whether there's a way to do it without two scrollbars. It's a bit aesthetically displeasing having so many scrollbars.

**Comment by ivansg44** [comment]
For some reason, I'm no longer experiencing this bug. When the rows re-size on scrolling, the frozen column line up with the rest of the columns. @cmrn-rhi Are you able to replicate this bug with the latest release?

**Comment by ddooley** [comment]
It may be fixed due to the newer rendering configuration you did, Ivan.

**Comment by cmrn-rhi** [comment]
That's great! I'll give it a try when I get home this weekwnd.

**Comment by cmrn-rhi** [comment]
I found it was still happening when testing using the master branch if a cell in the frozen column has text that wraps around to a new line:

![DataHarmonizer-Issue66_02](https://user-images.githubusercontent.com/48695054/89719624-0ff4d900-d97f-11ea-9929-02964e3dca00.png)

**Comment by ivansg44** [comment]
I was able to duplicate it. This will remain an open issue.

**Comment by griffie** [comment]
I saw this too. Mainly when there's lots of data entered.

**Comment by ddooley** [comment]
I think this is an error that would best be reported back to the https://handsontable.com/ folks.  I would think its showing up for their other clients/community users too. But first, we'd need to update to Handsontable 8.3.2 (https://github.com/handsontable/handsontable/issues/7637 ) to test if that solves the problem.

**Comment by ddooley** [comment]
I tried adding afterScrollVertically() to master DH main.js createHot call (and tested autoColumnSize and autoRowSize true/false attributes), but still the rendering problem came up.  We were running handsontable 7.4.2 , release date 19/02/2020.  I guess I can take it up with the them.

**Comment by ddooley** [comment]
Note, a temporary usability workaround is to highlight the row one is interested in; this highlight remains visible even if row miss-alignment occurs.

**Comment by ddooley** [comment]
I think this is fixed now in latest master 0.15.0 .  I have made a main.css stylesheet change which seem to have fixed the horizontal and vertical scroll problem misalignment with first frozen column. The nowrap and ellipses configuration seems to prevent miscalculation.

```
#grid table.htCore > tbody > tr td:nth-child(2) {
	border-right: medium solid gray;
  	white-space: nowrap;
	text-overflow: ellipsis;
}
```

As well I turned on the handsontable feature (manualColumnResize: true) that allows users to set widths of columns, which kicks in AFTER column widths have been calculated on data load.
